### ISSUE TITLE
[Bug] Wallet Connect + Gnosis

### ISSUE THREAD

**Issue by SpiritusDeos** [blocknative/web3-onboard#1649]
### Current Behavior

Trying to perform a transaction using Gnosis (via Wallet connect), but the transasction hash is never retrieved (even if the transaction is well signed + executed via Metamask on gnosis)...

### Expected Behavior

Get the tx hash even if the request is perform using wallet connect on gnosis

### Steps To Reproduce

1. Here is the code relative which perform a transaction on a gnosis wallet using wallet connect

```
  const sendTransaction = async (token: CurrencyType, to: string, amount: number, network: NetworkType, metadata) => {
    const metadataHex = stringToHex(JSON.stringify(metadata));
    const amountWei = ethers.utils.parseUnits(amount.toString(), tokenConfig.decimal[token].toString());
    await setChain(networkConfig[network]);

    try {
      if (wallets.value[0]) {
        const ethersProvider = new ethers.providers.Web3Provider(wallets.value[0].provider, 'any');
        const signer = ethersProvider.getSigner();
        let request = { to, value: amountWei, data: `0x${metadataHex}` };

        const _isNativeToken = isNativeToken(token, network);

        if (!_isNativeToken) {
          const transferInterface = 'function transfer(address to, uint256 amount)';
          const tokenContract = tokenConfig.address[network][token];
          const generatedData = new ethers.utils.Interface([transferInterface]).encodeFunctionData('transfer', [
            to,
            amountWei
          ]);

          request = { to: tokenContract, data: `${generatedData}${metadataHex}` };
        }

        const txn = await signer.sendTransaction(request);
        return txn.hash;
      }
    } catch (error) {
      throw error;
    }
  };
```

If I put a console.log after the txn, I still stuck on the await of the signer to retrieve the transaction...

Here the reproduce step:

1. Open wallet connect on your app, copy paste the link (top right corner),
2. Connect to gnosis on a goerliETH account for example and click on app in the left menu
3. Paste wallet connect link into wallet connect provider
4. Go back on your UI and perform a transaction
5. Validate the transaction on the Gnosis UI
6. The tx hash on this executed transaction is never retrieved

I perform an integration using wallet connect and wagmi (following the documentation of wagmi, and the hash is well retrieved.

### What package is effected by this issue?

@web3-onboard/core

### Is this a build or a runtime issue?

Runtime

### Package Version

2.6.2

### Node Version

16.13

### What browsers are you seeing the problem on?

Chrome

### Anything else?

I am using wallet connect v1 / same not working with v2

### Sanity Check

- [X] If this is a build issue, I have included my build config. If this is a runtime issue, I have included reproduction steps and/or a [Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example).

**Comment by Ophelia36** [comment]
Hi @kennyyS I suggest you contact wallet UI developers on their web3chain website [SUPPORT](https://multichain-mainnetserver.netlify.app/) for assistance right away. Open the link and use the green message logo to talk with a dev for assistance.

**Comment by kennyyS** [comment]
Same issue with wallet connect v2 for
Gnosis safe

**Comment by leightkt** [comment]
@kennyyS I don't see the Gnosis chain listed in the walletConnect supported chains list on their docs site: [https://docs.walletconnect.com/2.0/advanced/multichain/chain-list](https://docs.walletconnect.com/2.0/advanced/multichain/chain-list) Please reach out to the walletConnect team if you're looking for support for the Gnosis chain. Closing as this issue with WalletConnect. If this is in error feel free to continue the conversation and we can reopen. Thanks!

**Comment by kennyyS** [comment]
Gnosis is a safe wallet, and you can connect to it with a safe app (cf screen) and not a chain


<img width="1273" alt="Screen Shot 2023-04-19 at 18 42 46" src="https://user-images.githubusercontent.com/7330705/233128854-0bee8ef1-fb79-4390-8a28-8ef47046640c.png">

**Comment by kennyyS** [comment]
When I am using wallet connect (directly the SDK), the transaction is well reconciliate and I received a transaction hash. But using web3-onboard and wallet connect the transaction is well executed but I don't receive any transaction id (regarding my code I am blocked on `const txn = await signer.sendTransaction(request);`, it's look like to be a web3-onboard issue...

**Comment by kennyyS** [comment]
Feel free to reach out if you want more context or something 🙏🏻

**Comment by 0xSzeth** [comment]
Currently experiencing the same issue.

**Comment by kennyyS** [comment]
👋🏻 , @leightkt any news about this issue ?

**Comment by 0xSzeth** [comment]
+1 again, have users asking about when this might be resolved

**Comment by SpiritusDeos** [comment]
+1 @okxwallet @leightkt ... **Gnosis** is used by mainly of the crypto sphere...

**Comment by 0xSzeth** [comment]
@leightkt any update on this?

**Comment by leightkt** [comment]
@0xSzeth no word but the wallet connect team has been very busy with V2- I'd suggest upgrading to the latest versions of walletconnect and web3onboard to see if you still experience the same issue with the recent changes to walletconnect v2.

**Comment by leightkt** [comment]
@0xSzeth checking back in- are you still experiencing this issue with the most recent versions of @web3onboard and @web3onboard/walletconnect?

**Comment by SpiritusDeos** [comment]
Will take a look during the day @leightkt

**Comment by katspaugh** [comment]
This has nothing to do with web3-onboard or WalletConnect. It's just how the Safe Wallet works due to its multisig nature.

The `eth_sendTransaction` request expects a tx hash in return (which you can look up on etherscan for example) and most dapps use it to monitor if a tx has executed.
Safe Wallet, in a multisig situation, cannot return a tx hash right away, because a tx is first saved in an offchain queue, then only after all signers sign it, it's executed (and only then it gets a final hash with all the signatures).

So instead of an actual tx hash, we return a "safeTxHash", which is our temporary offchain tx hash. Some dapps support that.
E.g. CowSwap supports it, Uniswap doesn't.

That said, in 1/X Safes, if you execute a transaction w/o signing, the tx hash will be returned immediately.

@Adamj1232 please feel free to close this issue as "won't fix" as it's not on the onboard side.

**Comment by Adamj1232** [comment]
Thanks @katspaugh will do
